tests/topotests/rip_bfd_topo1/test_rip_bfd_topo1.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 eval: (blacken-mode 1) -*-
+#
+# test_rip_bfd_topo1.py
+# Part of NetDEF Topology Tests
+#
+# Copyright (c) 2023 by
+# Network Device Education Foundation, Inc. ("NetDEF")
+#
+# Permission to use, copy, modify, and/or distribute this software
+# for any purpose with or without fee is hereby granted, provided
+# that the above copyright notice and this permission notice appear
+# in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND NETDEF DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL NETDEF BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+# ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+#
+
+"""
+test_rip_bfd_topo1.py: Test RIP BFD integration.
+"""
+
+import sys
+import re
+import pytest
+
+from functools import partial
+from lib import topotest
+from lib.topogen import Topogen, TopoRouter
+from lib.topolog import logger
+
+pytestmark = [
+    pytest.mark.bfdd,
+    pytest.mark.ripd,
+]
+
+
+@pytest.fixture(scope="module")
+def tgen(request):
+    "Setup/Teardown the environment and provide tgen argument to tests"
+
+    topodef = {
+        "s1": ("r1", "r2"),
+        "s2": ("r1", "r3")
+    }
+    tgen = Topogen(topodef, request.module.__name__)
+    tgen.start_topology()
+
+    router_list = tgen.routers()
+
+    for router_name, router in router_list.items():
+        router.load_config(TopoRouter.RD_BFD, "bfdd.conf")
+        router.load_config(TopoRouter.RD_RIP, "ripd.conf")
+        router.load_config(TopoRouter.RD_ZEBRA, "zebra.conf")
+        if router_name in ["r2", "r3"]:
+            router.load_config(TopoRouter.RD_STATIC, "staticd.conf")
+
+    tgen.start_router()
+    yield tgen
+    tgen.stop_topology()
+
+
+@pytest.fixture(autouse=True)
+def skip_on_failure(tgen):
+    "Test if routers is still running otherwise skip tests"
+    if tgen.routers_have_failure():
+        pytest.skip("skipped because of previous test failure")
+
+
+def show_rip_json(router):
+    "Get router 'show ip rip' JSON output"
+    output = router.vtysh_cmd("show ip rip")
```

Comment:
This can be fetched using `show yang operational-data` already?

Reply:
Unfortunately no, the yang operational data doesn't have the route type and I'm not sure how it displays ECMP routes (the test checks for multipath route being changed)